src/utils/utils.hpp

```diff
@@ -497,7 +497,7 @@ namespace detail {
         typename T,
         typename D
         // workaround a msvc bug https://developercommunity.visualstudio.com/t/MSVC-1938331290-preview-fails-to-comp/10505565
-        #if !defined(_MSC_VER) || _MSC_VER != 1938
+        #if !defined(_MSC_VER) || (_MSC_VER != 1900 && _MSC_VER != 1938) // TODO: Are you sure only these versions are affacted?
```

Comment:
This was a 19.38-specific bug however it appears <=19.23 fails for a different reason https://godbolt.org/z/6Y5EvdWPK

Reply:
Thanks for the godbolt link, how should I proceed?
It seems like a lot of versions fail.

Reply:
`#if !defined(_MSC_VER) || !(_MSC_VER <= 1923 || _MSC_VER == 1938)` should do the trick. Could you do me a favor and add a note above about why <=19.23 is disabled here? (the compiler explorer link would be good to include)